contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/MapRDBPushProjectIntoScan.java

```diff
@@ -0,0 +1,145 @@
+    super(operand, description);
+  }
+
+  public static final StoragePluginOptimizerRule PROJECT_ON_SCAN = new MapRDBPushProjectIntoScan(
+      RelOptHelper.some(ProjectPrel.class, RelOptHelper.any(ScanPrel.class)), "MapRDBPushProjIntoScan:Proj_On_Scan") {
+    @Override
+    public void onMatch(RelOptRuleCall call) {
+      final ScanPrel scan = (ScanPrel) call.rel(1);
+      final ProjectPrel project = (ProjectPrel) call.rel(0);
+      if (!(scan.getGroupScan() instanceof MapRDBGroupScan)) {
+        return;
+      }
+      doPushProjectIntoGroupScan(call, project, scan, (MapRDBGroupScan) scan.getGroupScan());
+      if (scan.getGroupScan() instanceof BinaryTableGroupScan) {
+        BinaryTableGroupScan groupScan = (BinaryTableGroupScan) scan.getGroupScan();
+
+      } else {
+        assert (scan.getGroupScan() instanceof JsonTableGroupScan);
+        JsonTableGroupScan groupScan = (JsonTableGroupScan) scan.getGroupScan();
+
+        doPushProjectIntoGroupScan(call, project, scan, groupScan);
+      }
+    }
+
+    @Override
+    public boolean matches(RelOptRuleCall call) {
+      final ScanPrel scan = (ScanPrel) call.rel(1);
+      if (scan.getGroupScan() instanceof BinaryTableGroupScan ||
+          scan.getGroupScan() instanceof JsonTableGroupScan) {
+        return super.matches(call);
+      }
+      return false;
+    }
+  };
+
+  protected void doPushProjectIntoGroupScan(RelOptRuleCall call,
+      ProjectPrel project, ScanPrel scan, MapRDBGroupScan groupScan) {
+    try {
+
+      DrillRelOptUtil.ProjectPushInfo columnInfo =
+          DrillRelOptUtil.getFieldsInformation(scan.getRowType(), project.getProjects());
+      if (columnInfo == null || Utilities.isStarQuery(columnInfo.getFields()) //
+          || !groupScan.canPushdownProjects(columnInfo.getFields())) {
+        return;
+      }
+      RelTraitSet newTraits = call.getPlanner().emptyTraitSet();
+      // Clear out collation trait
+      for (RelTrait trait : scan.getTraitSet()) {
+        if (!(trait instanceof RelCollation)) {
+          newTraits.plus(trait);
+        }
+      }
+      final ScanPrel newScan = new ScanPrel(scan.getCluster(), newTraits.plus(Prel.DRILL_PHYSICAL),
+          groupScan.clone(columnInfo.getFields()),
+          columnInfo.createNewRowType(project.getInput().getCluster().getTypeFactory()), scan.getTable());
+
+      List<RexNode> newProjects = Lists.newArrayList();
+      for (RexNode n : project.getChildExps()) {
+        newProjects.add(n.accept(columnInfo.getInputReWriter()));
+      }
+
+      final ProjectPrel newProj =
+          new ProjectPrel(project.getCluster(),
+              project.getTraitSet().plus(Prel.DRILL_PHYSICAL),
+              newScan,
+              newProjects,
+              project.getRowType());
+
+      if (ProjectRemoveRule.isTrivial(newProj) &&
+          // the old project did not involve any column renaming
+          sameRowTypeProjectionsFields(project.getRowType(), newScan.getRowType())) {
```

Comment:
This may have to do with the fact that `MapRDBPushProjectIntoScan` is applied during physical planning unlike the `DrillPushProjectIntoScanRule` so there may be some differences in how the pushdown occurs.  We would have added the extra check `sameRowTypeProjectionsFields` to prevent an incorrect pushdown, although I don't recall exact details now.